plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java

```diff
@@ -14,21 +14,61 @@
 package io.trino.plugin.elasticsearch;
 
 import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
 import org.apache.http.entity.ContentType;
 import org.apache.http.nio.entity.NStringEntity;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
 import java.io.IOException;
 
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryRunner.createElasticsearchQueryRunner;
 import static java.lang.String.format;
 
 public class TestElasticsearch6ConnectorTest
         extends BaseElasticsearchConnectorTest
 {
+    private final ElasticsearchServer elasticsearch;
+    private final RestHighLevelClient client;
+
     public TestElasticsearch6ConnectorTest()
     {
-        super("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0");
+        try {
+            elasticsearch = new ElasticsearchServer("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0", ImmutableMap.of());
+            client = elasticsearch.createClient();
+        }
+        catch (IOException e) {
+            throw new RuntimeException("Cannot build ElasticsearchServer", e);
+        }
+    }
+
+    @Override
+    protected RestHighLevelClient getClient()
+    {
+        return client;
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createElasticsearchQueryRunner(elasticsearch.getAddress(), TpchTable.getTables(), ImmutableMap.of(), ImmutableMap.of(), 3);
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    public final void destroy()
+    {
+        try {
+            elasticsearch.stop();
```

Comment:
You can use `closeAfterClass` in `createQueryRunner` so destroy method might not be needed.

Reply:
Applied